xbmc/pvr/epg/EpgContainer.h

```diff
@@ -225,6 +226,34 @@ namespace PVR
      */
     int CleanupCachedImages();
 
+    /*!
+     * @brief Get all saved searches from the database.
+     * @param bRadio Whether to fetch saved searches for radio or TV.
+     * @return The searches.
+     */
+    std::vector<std::shared_ptr<CPVREpgSearchFilter>> GetSavedSearches(bool bRadio);
+
+    /*!
+     * @brief Persist a saved search in the database.
+     * @param search The saved search.
```

Comment:
Maybe change to `@param[in,out] search The saved search.` to know why not const and to add a small note what becomes edited.

Found this as edit:
- `epgSearch.SetDatabaseId(static_cast<int>(m_pDS->lastinsertid()));`
- `epgSearch.SetChanged(false);`

And the read of CPVREpgSearchFilter values from it.

Reply:
Fair point, and if this pay for you I would prefer to change this throughout the whole pvr module in a follow up PR, not in this PR.